mobify/platform-scaffold
# Non-PWA mode support

Changes to support non-PWA mode

 **JIRA**: https://mobify.atlassian.net/browse/MSG-515
 **Linked PRs**:
- #719 
- https://github.com/mobify/progressive-web-sdk/pull/749
- https://github.com/mobify/duncan/pull/86

## Changes
- Change `loader.js` - if not running in a supported PWA browser, for a PWA route, then if the browser supports 'non-PWA' mode, load the worker, messaging client and non-PWA script, and init.
- Implementation of basic non-pwa Messaging in `web/non-pwa/non-pwa.js`

## How to test-drive this PR
- First, run `npm install` to install the `html-loader` dependency
- Check out this branch of the SDK: https://github.com/mobify/progressive-web-sdk/pull/749
- Inside the SDK, run `npm link` then `npm run dev:build`
- Inside the scaffold, run `npm link progressive-web-sdk`
- Follow the instructions in https://github.com/mobify/platform-scaffold/pull/719 (Charles setup for V8 tag)
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1) but **with a desktop-based user agent** (i.e. no mobile emulation!)
- You should see a soft ask UI appear from the bottom of the screen
- Selecting 'Yes Please' should trigger a notification permission system-ask dialog
- Select 'No Thanks' should dismiss the UI. Going to a new page should _not_ redisplay the UI
